modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/authenticator/LdapAuthenticator.java

```diff
@@ -0,0 +1,98 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package io.ballerina.messaging.broker.auth.authentication.authenticator;
+
+import io.ballerina.messaging.broker.auth.AuthException;
+import io.ballerina.messaging.broker.auth.BrokerAuthConfiguration;
+import io.ballerina.messaging.broker.auth.authentication.AuthResult;
+import io.ballerina.messaging.broker.auth.authentication.Authenticator;
+import io.ballerina.messaging.broker.auth.authorization.UserStore;
+import io.ballerina.messaging.broker.auth.ldap.LdapAuthHandler;
+import io.ballerina.messaging.broker.common.StartupContext;
+import io.ballerina.messaging.broker.common.config.BrokerConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+
+/**
+ * Ldap authentication representation for @{@link Authenticator}.
+ */
+public class LdapAuthenticator implements Authenticator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapAuthenticator.class);
+
+    private BrokerAuthConfiguration.LdapConfiguration ldapConfiguration;
+    private LdapAuthHandler ldapAuthHandler;
+
+    @Override
+    public void initialize(StartupContext startupContext,
+                           UserStore userStore,
+                           Map<String, Object> properties) throws Exception {
+
+        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
+        BrokerAuthConfiguration brokerAuthConfiguration = configProvider.getConfigurationObject(
+                BrokerAuthConfiguration.NAMESPACE, BrokerAuthConfiguration.class);
+        ldapConfiguration = brokerAuthConfiguration.getAuthentication().getAuthenticator().getLdap();
+
+        if (Objects.isNull(ldapConfiguration)) {
+            throw new Exception("Error in ldap configuration");
+        }
+
+        ldapAuthHandler = new LdapAuthHandler(ldapConfiguration);
+    }
+
+    @Override
+    public AuthResult authenticate(String username, char[] password) throws AuthException {
+
+        boolean isAuthenticated = false;
+
+        DirContext dirContext = null;
+        try {
+            dirContext = ldapAuthHandler.connectAnonymously();
+        } catch (NamingException e) {
+            throw new AuthException("Ldap connection failed", e);
+        }
+
+        String dn = null;
+        try {
+            dn = ldapAuthHandler.searchDN(dirContext, username);
+        } catch (NamingException e) {
+            throw new AuthException("Error while searching Username: " + username, e);
+        } finally {
+            if (dirContext != null) {
+                try {
+                    dirContext.close();
+                } catch (NamingException e) {
```

Comment:
Why are we leaving the catch block empty?

Reply:
Its not important whether an exception occurred here or not. this is just closing a resource.

Reply:
Shall we log the error?

Reply:
ok